core/src/main/scala/org/apache/spark/deploy/master/ui/ApplicationPage.scala

```diff
@@ -77,7 +77,15 @@ private[ui] class ApplicationPage(parent: MasterWebUI) extends WebUIPage("app")
             <li><strong>State:</strong> {app.state}</li>
             {
               if (!app.isFinished) {
-                <li><strong><a href={app.desc.appUiUrl}>Application Detail UI</a></strong></li>
+                <li><strong>
+                  {
+                    if (parent.master.reverseProxy) {
+                      <a href={"/target/" + app.id + "/"}>Application Detail UI</a>
```

Comment:
By all means it can be a configuration value, but there were already some literal conventions eg. "app" so I thought of having it this way. But if community prefer the configured value, then I will add that.